core/offchain.go

```diff
@@ -39,7 +39,6 @@ func (bc *BlockChain) CommitOffChainData(
 	isStaking := bc.chainConfig.IsStaking(block.Epoch())
 	isPreStaking := bc.chainConfig.IsPreStaking(block.Epoch())
 	header := block.Header()
-	isNewEpoch := len(header.ShardState()) > 0
```

Comment:
better keep the isNewEpoch variable so not repeated call IsLastBlockInEpoch(). Also less code change.

Reply:
Sure changed back.